src/rust/engine/src/tasks.rs

```diff
@@ -45,43 +45,106 @@ impl rule_graph::Rule for Rule {
+          rule_type,
+          task_name,
+          clause_portion,
+          product,
+          get_portion,
+        } = Self::extract_task_elements(task, 2, 1);
+
+        format!(
+          "@{}({}) -> {}{}\n{}",
+          rule_type, clause_portion, product, get_portion, task_name,
+        )
+      }
+      Rule::Intrinsic(ref intrinsic) => format!(
+        "@rule(<intrinsic>({}) -> {})",
+        intrinsic.input, intrinsic.product,
+      ),
+    }
+  }
+}
+
+struct TaskElements {
+  rule_type: String,
+  task_name: String,
+  clause_portion: String,
+  product: String,
+  get_portion: String,
+}
+
+impl Rule {
+  fn extract_task_elements(
+    task: &Task,
+    select_clause_threshold: usize,
+    get_clause_threshold: usize,
+  ) -> TaskElements {
+    let product = format!("{}", task.product);
+    let select_clauses = task
+      .clause
+      .iter()
+      .map(|c| c.product.to_string())
+      .collect::<Vec<_>>();
+    let clause_portion = if select_clauses.len() <= select_clause_threshold {
+      select_clauses.join(", ").to_string()
+    } else {
+      format!("\n{},\n", select_clauses.join(",\n"))
+    };
+    let get_clauses = task
+      .gets
+      .iter()
+      .map(::std::string::ToString::to_string)
+      .collect::<Vec<_>>();
+    let rule_type = if task.cacheable {
+      "rule".to_string()
+    } else {
+      "goal_rule".to_string()
+    };
+    let get_portion = if get_clauses.is_empty() {
+      "".to_string()
+    } else if get_clauses.len() <= get_clause_threshold {
+      format!(",\ngets=[{}]", get_clauses.join(", "))
+    } else {
+      format!(",\ngets=[\n{},\n]", get_clauses.join("\n"))
+    };
+
+    TaskElements {
+      rule_type,
+      task_name: task.func.name(),
+      clause_portion,
+      product,
+      get_portion,
+    }
+  }
 }
 
 impl fmt::Display for Rule {
   fn fmt(&self, f: &mut fmt::Formatter) -> Result<(), fmt::Error> {
     match self {
       &Rule::Task(ref task) => {
-        let product = format!("{}", task.product);
-        let params = task
-          .clause
-          .iter()
-          .map(|c| c.product.to_string())
-          .collect::<Vec<_>>()
-          .join(", ");
-        let get_portion = if task.gets.is_empty() {
-          "".to_string()
-        } else {
-          let get_members = task
-            .gets
-            .iter()
-            .map(::std::string::ToString::to_string)
-            .collect::<Vec<_>>()
-            .join(", ");
-          format!(", gets=[{}]", get_members)
-        };
-        let rule_type = if task.cacheable {
-          "rule".to_string()
-        } else {
-          "goal_rule".to_string()
-        };
-        write!(
-          f,
-          "@{}({}({}) -> {}{})",
+        let TaskElements {
           rule_type,
-          task.func.name(),
-          params,
+          task_name,
+          clause_portion,
           product,
           get_portion,
+        } = Self::extract_task_elements(task, 7, 5);
```

Comment:
Due to soft wrapping done by terminals, I think we don't ever want hardcoded newlines in this `fmt::Display` method. Consider changing `::extract_task_elements` to take `Optional<int>` for the threshold parameter and passing `None` here.